src/ui/public/vis/editors/default/vis_type_field_filter.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { AggType } from '../../../agg_types';
+import { FieldParamType } from '../../../agg_types/param_types';
+import { aggTypeFieldFilters } from '../../../agg_types/param_types/filter';
+import { IndexPattern } from '../../../index_patterns';
+import { AggConfig } from '../../../vis';
+
+import { propFilter } from '../../../filters/_prop_filter';
+
+const filterByType = propFilter('type');
+
+/**
+ * This filter checks the defined aggFilter in the schemas of that visualization
+ * and limits available aggregations based on that.
+ */
+aggTypeFieldFilters.addFilter(
+  (
+    field: any,
+    aggType: AggType,
+    fieldParamType: FieldParamType,
+    indexPattern: IndexPattern,
+    aggConfig: AggConfig
+  ) => {
+    if (
+      (fieldParamType.onlyAggregatable && !field.aggregatable) ||
+      (!fieldParamType.scriptable && field.scripted)
+    ) {
+      return false;
+    }
+
+    if (fieldParamType.filterFieldTypes) {
```

Comment:
Maybe use destructuring here to reduce repetition, and use an early exit so that the bulk of the work is done at the end of the function? We could also use a ternary to make the binary assignment of `filters` clearer.

```js
    const { filterFieldTypes } = fieldParamType;
    
    if (!filterFieldTypes) {
      return true;
    }

    let filters = _.isFunction(filterFieldTypes)
      ? filterFieldTypes.bind(this, aggConfig.vis)
      : filterFieldTypes;

    return filterByType([field], filters).length !== 0;
```

Reply:
restructured this function with this feedback 🙂